dotnet/src/Agents/Bedrock/BedrockAgent.cs

```diff
@@ -117,11 +118,18 @@ public override async IAsyncEnumerable<AgentResponseItem<ChatMessageContent>> In
             () => new BedrockAgentThread(this.RuntimeClient),
             cancellationToken).ConfigureAwait(false);
 
+        // Get the context contributions from the AIContextBehaviors.
+#pragma warning disable SKEXP0110 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+        var extensionsContext = await bedrockThread.AIContextBehaviors.OnModelInvokeAsync(messages, cancellationToken).ConfigureAwait(false);
+#pragma warning restore SKEXP0110 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+
         // Ensure that the last message provided is a user message
         string? message = this.ExtractUserMessage(messages.Last());
 
-        // Build session state with conversation history if needed
+        // Build session state with conversation history and override instructions if needed
         SessionState sessionState = this.ExtractSessionState(messages);
+        var mergedAdditionalInstructions = MergeAdditionalInstructions(options?.AdditionalInstructions, extensionsContext.Instructions);
+        sessionState.PromptSessionAttributes = new() { ["AdditionalInstructions"] = mergedAdditionalInstructions };
```

Comment:
"AdditionalInstructions" is a good candidate for a `const string` in the `PromptSessionAttributes` and potentially an `internal` ctor.



```suggestion
        sessionState.PromptSessionAttributes = new(mergedAdditionalInstructions);
```

Reply:
Sure, moved to a constant.